### PR TITLE
Grafana dashboard for trace quality

### LINKED ISSUE #16: Grafana dashboard for trace quality metrics
Provide a Grafana dashboard for trace quality metrics. Preferably a dashboard per service.

* visualize each individual metric
* calculate a single KPI e.g. 1.0 if all checks are passing

### PR DESCRIPTION
**Pull request by pavolloffay**
Resolves #16 

@annanay25 could you please have a look? 

Signed-off-by: Pavol Loffay <ploffay@redhat.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -33,3 +33,8 @@ hotrod-run:
 prom-run:
 	echo "Open browser on :9090"
 	docker run --rm --net=host -v ${PWD}/manifests/prometheus-config.yml:/etc/prometheus/prometheus.yml prom/prometheus:latest
+
+.PHONY: grafana-run
+grafana-run:
+	echo "Open browser on :3000"
+	docker run --rm -it --net=host grafana/grafana
```
Comment (annanay25) [inline Makefile]: Small nit: A config file and the dashboard.json can be mounted on the container directly!

Ref - https://grafana.com/docs/grafana/latest/administration/provisioning/#dashboards

File: dashboards/Grafana-tracequality.json [hunk 2]
```diff
@@ -0,0 +1,476 @@
+{
```
Comment (annanay25) [inline dashboards/Grafana-tracequality.json]: Dashboard creation can be templated through the mixin (we can use it for future releases if dashboards get bigger) 
Ref- https://github.com/jaegertracing/jaeger/pull/1668

**Comment by pavolloffay** [comment]
@annanay25 the dashboard has one configurable variable - service. Do you know if grafana can be configured to search possible values of this variable based on the Prometheus metric label?

Screenshot of Trace quality - Grafana (1)

File: dashboards/Grafana-tracequality.json (hunk 2 above)
Comment (pavolloffay) [reply dashboards/Grafana-tracequality.json]: +1 I think we could have more dashboards.

A dashboard I am thinking about would show other derived metrics from traces per service. We could also define alerts for these metrics.

**Comment by annanay25** [comment]
> Do you know if grafana can be configured to search possible values of this variable based on the Prometheus metric label?

@pavolloffay - Yes, this is definitely possible.
Something like `$service = label_name("trace_quality_server_tag_total", "service")` should do the trick.

References -
https://grafana.com/docs/grafana/latest/features/datasources/prometheus/#query-variable
https://banzaicloud.com/blog/grafana-templating/

File: dashboards/Grafana-tracequality.json (hunk 2 above)
Comment (annanay25) [reply dashboards/Grafana-tracequality.json]: I'm not sure if users would need alerts on these metrics, but definitely +1 for dashboards.

File: Makefile (hunk 1 above)
Comment (pavolloffay) [reply Makefile]: I have added a config file for datasource and dashboards. In the tracequality dashboard I hardcoded datasource `Prometheus`. We can change this once we introduce mixins.

**Comment by pavolloffay** [comment]
Am I doing something wrong? `label_values("trace_quality_server_tag_total", "service")` - I am interested to get values of the service label

Screenshot of Trace quality - Grafana (2)

**Comment by pavolloffay** [comment]
`label_values("service")` didn't work for me either

**Comment by pavolloffay** [comment]
I will merge this and create issue to populate label values to service dropdown.

**Comment by annanay25** [comment]
Strange, I'll try getting it working on my local machine and get back

**Pull request merged**